content/08-mongodb/10-mongodb-transactions.mdx

```diff
@@ -0,0 +1,246 @@
+
+In this guide, we'll begin by discussing what transactions are, when to use them in the document model, and how they work conceptually in MongoDB.
+
+<PrismaOutlinks>
+
+If you are using [Prisma Client](https://www.prisma.io/docs/concepts/components/prisma-client) with MongoDB, you can use the [MongoDB connector (currently in preview)](https://www.prisma.io/docs/concepts/database-connectors/mongodb) to connect and manage your data.
+
+</PrismaOutlinks>
+
+
+## What are transactions?
+
+Transactions are a way to group together and isolate multiple statements for processing as a single operation. Instead of executing each command individually as it is sent to the server, in a transaction, commands are bundled together and executed in a separate context than other requests.
+
+Isolation is an important part of transactions. Within a transaction, the executed statements can only affect the environment within the transaction itself. From inside the transaction, statements can modify data and the results are immediately visible. From the outside, no changes are made until the transaction is committed, at which time all of the actions within the transaction become visible at once.
+
+These features help databases achieve [ACID compliance](https://www.prisma.io/dataguide/dataguide/datamodeling/in-vivo) by providing atomicity (actions in a transaction are either all committed or all rolled back) and isolation (outside of the transaction, nothing changes until the commit while inside, the statements have consequences). These together help the database maintain consistency (by guaranteeing that partial data transformations cannot occur). Furthermore, changes in transactions are not returned as successful until they are committed to non-volatile storage, which provides durability.
+
+MongoDB has support for [multi-document ACID transactions](https://www.mongodb.com/blog/post/mongodb-multi-document-acid-transactions-general-availability) and [distributed multi-document ACID transactions](https://www.mongodb.com/press/mongodb-42-adds-distributed-transactions-field-level-encryption-updated-kubernetes-operator-and-more-to-the-leading-modern-general-purpose-database). By nature, the document model allows related data to be stored together in a single document. The document model, combined with atomic document updates, takes away the need for transactions in a majority of use cases. However, there are cases where multi-document, multi-collection MongoDB transactions are your best choice.
+
+## When to use transactions in the document model
+
+The document model inherently solves for a lot of the needs that transactions target. Nonetheless, there are some use cases that stand out for their need to utilize transactions even in the document model.
```

Comment:
"The document model inherently solves for a lot of the needs that transactions target."  It might be helpful to include a basic explanation of how that works.  Like what specifically about the document model helps avoid the scenarios that transactions are designed to fix.